yt_dlp/YoutubeDL.py

```diff
@@ -1354,8 +1354,16 @@ def create_key(outer_mobj):
                     value, default = None, na
 
             fmt = outer_mobj.group('format')
-            if fmt == 's' and last_field in field_size_compat_map and isinstance(value, int):
-                fmt = f'0{field_size_compat_map[last_field]:d}d'
+            # Apply numeric padding for both direct numeric fields and fields with replacements
+            if last_field in field_size_compat_map and isinstance(value, (int, str)):
+                # Try to convert string values (from replacements) back to int
+                try:
+                    if isinstance(value, str):
+                        value = int(value)
+                    if isinstance(value, int):
+                        fmt = f'0{field_size_compat_map[last_field]:d}d'
+                except (ValueError, TypeError):
+                    pass
```

Comment:
This doesn't work, because at this point in the code we have already lost track of what the original value was and only have access to the full replacement value